### PR TITLE
home work #8

### PR DESCRIPTION
**Pull request by DeliriumInTheVoid**

### CODE REVIEW

File: lesson_08/cpp/cyclic_functions.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#include "../headers/cyclic_functions.hpp"
+
+void count_numbers_by_sign()
+{
+	int numbers_qty{ 0 }, positive_qty{ 0 }, negative_qty{ 0 }, zero_qty{ 0 };
+
+	std::cout << "Enter numbers quantity:\n";
+	std::cin >> numbers_qty;
+
+	while (--numbers_qty >= 0)
```
Comment (bohdanpc) [inline lesson_08/cpp/cyclic_functions.cpp:10]: Prefer `for` loop for simple iteration with index

File: lesson_08/cpp/cyclic_functions.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	else
+	{
+		long long current_value = 1;
+
+		for (int i = 2; i <= count; i++)
+		{
+			current_value *= i;
+		}
+
+		result = current_value;
```
Comment (bohdanpc) [inline lesson_08/cpp/cyclic_functions.cpp:48]: just make early `return current_value;`

File: lesson_08/cpp/cyclic_functions.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	std::cout << "Positive: " << positive_qty << "; Negative: " << negative_qty << "; Zero: " << zero_qty << std::endl;
+}
+
+long long factorial_cycle(const int count)
+{
+	long long result{ 0 };
+
+	if (count == 0 || count == 1)
+	{
+		result = 1;
```
Comment (bohdanpc) [inline lesson_08/cpp/cyclic_functions.cpp:37]: `return` here as well and you won't need `result` tmp variable at all making your code shorter and cleaner

File: lesson_08/cpp/cyclic_functions.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	}
+	else
```
Comment (bohdanpc) [inline lesson_08/cpp/cyclic_functions.cpp:39]: and with early returns `else` won't be needed too

File: lesson_08/headers/recursive_functions.hpp [hunk 5]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#include <iostream>
```
Comment (bohdanpc) [inline lesson_08/headers/recursive_functions.hpp:3]: don't include in .h/.hpp files headers you don't actually need for this header to work properly.

**Review by bohdanpc — CHANGES_REQUESTED** [review]

File: lesson_08/cpp/cyclic_functions.cpp (hunk 1 above)
Comment (DeliriumInTheVoid) [reply lesson_08/cpp/cyclic_functions.cpp:10]: Fixed. In this case with "while()" I do not have useless "i" variable.

File: lesson_08/headers/recursive_functions.hpp (hunk 5 above)
Comment (DeliriumInTheVoid) [reply lesson_08/headers/recursive_functions.hpp:3]: fixed

File: lesson_08/cpp/cyclic_functions.cpp (hunk 3 above)
Comment (DeliriumInTheVoid) [reply lesson_08/cpp/cyclic_functions.cpp:37]: Fixed. Can influence on cyclomatic complexity metric.

File: lesson_08/cpp/cyclic_functions.cpp (hunk 4 above)
Comment (DeliriumInTheVoid) [reply lesson_08/cpp/cyclic_functions.cpp:39]: fixed

File: lesson_08/cpp/cyclic_functions.cpp (hunk 2 above)
Comment (DeliriumInTheVoid) [reply lesson_08/cpp/cyclic_functions.cpp:48]: fixed

**Pull request merged**